include/seqan3/alphabet/aminoacid/concept.hpp

```diff
@@ -0,0 +1,133 @@
+// ============================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ============================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Joshua Kim <joshua.kim AT fu-berlin.de>
+ * \brief Provides seqan3::aminoacid_concept.
+ */
+
+#pragma once
+
+#include <seqan3/alphabet/concept.hpp>
+
+// ============================================================================
+// concept
+// ============================================================================
+
+namespace seqan3
+{
+
+/*!\interface seqan3::aminoacid_concept <>
+ * \extends seqan3::alphabet_concept
+ * \brief A concept that indicates whether an alphabet represents amino acids.
+ * \ingroup aminoacid
+ *
+ * In addition to the requirements for seqan3::alphabet_concept, the aminoacid_concept introduces
+ * a requirement for an accepted alphabet. All aminoacid_concept members should accept the 27
+ * amino acid letters, and convert them to fit their respective size constraints.
```

Comment:
I would write instead:
```
In addition to the requirements for seqan3::alphabet_concept, the amino_acid_concept expects
conforming alphabets to provide an enum-like interface with all possible 27 amino acids as values
(although some may be mapped to others if the alphabet is smaller than 27).
```
Because it is more about requiring that the alphabet *provide* these than *accepting* these.